src/Control/Carrier/Choose/Church.hs

```diff
@@ -24,30 +23,32 @@ module Control.Carrier.Choose.Church
 , module Control.Effect.Choose
 ) where
 
-import           Control.Algebra
-import           Control.Applicative (liftA2)
-import           Control.Effect.Choose
-import qualified Control.Monad.Fail as Fail
```

Comment:
We were qualifying the import of `Control.Monad.Fail` everywhere when we really just needed to name it for compatibility with older `ghc`s. I’ve unqualified it everywhere, but still use its symbols under the `Fail.` prefix so that we don’t get redundant import warnings in newer `ghc`s.